src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaSerializationContext.cs

```diff
@@ -44,7 +44,7 @@ public override bool TryGetMessage<T>(out T message)
 
         public override bool TryGetMessage(Type messageType, out object message)
         {
-            if (messageType.IsAssignableFrom(typeof(TMessage)))
+            if (messageType.IsAssignableFrom(typeof(TMessage)) && _message.Value != null)
```

Comment:
I wonder if IsSupportedMessageType above needs to be adjusted as well to avoid issues with it being "supported" but not actually available.

Reply:
I adjusted, just in case